### PR TITLE
chore: fix github actions bot and unify labels

### PR DESCRIPTION
**Pull request by kacperkapusciak**
## Description

The goal of this PR is to unify labels throughout Software Mansion React Native Open Source projects.

This PR also simplifies cache in actions and updates the` bug-report.yml` issue template

### CODE REVIEW

**Review by tomekzaw — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -114,7 +114,7 @@ body:
     id: architecture
     attributes:
       label: Architecture
-      description: What React Native architecture your application is running on?
+      description: What React Native architecture your application is running on? Currently, the default architecture on React Native is Paper so if you haven't changed it your application select this option.
```
Comment (tomekzaw) [inline .github/ISSUE_TEMPLATE/bug-report.yml:117]: is this comment necessary?

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 2]
```diff
@@ -141,5 +141,16 @@ body:
   - type: input
     id: device-model
     attributes:
-      label: Real device model
-      placeholder: ex. Samsung Galaxy A22
+      label: Device model
+      description: On what device you are experiencing this problem? Specify full device name and the name and version of the operating system it's running.
```
Comment (tomekzaw) [inline .github/ISSUE_TEMPLATE/bug-report.yml:145]:
```suggestion
      description: What device you are experiencing this problem on? Specify full device name along with the version of the operating system it's running.
```

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 3]
```diff
@@ -141,5 +141,16 @@ body:
...
+
+  - type: dropdown
+    id: acknowledgements
+    attributes:
+      label: Acknowledgements
+      description: I searched for similar issues in the repository.
+      options:
+        - 'Yes'
+    validations:
+      required: true
```
Comment (tomekzaw) [inline .github/ISSUE_TEMPLATE/bug-report.yml:156]: Do we really need this?

Comment (kacperkapusciak) [reply .github/ISSUE_TEMPLATE/bug-report.yml:156]: triggered

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 1 above)
Comment (kacperkapusciak) [reply .github/ISSUE_TEMPLATE/bug-report.yml:117]: My reasoning is: 
When users enable Fabric they are well aware there are 2 architectures. Users that never heard of it before don't know there are many architectures.

Comment (tomekzaw) [reply .github/ISSUE_TEMPLATE/bug-report.yml:117]: ok lol

**Pull request merged**